### PR TITLE
chore: Update data sources to send FDv2-compatible payloads

### PR DESCRIPTION
**Pull request by keelerm84**
This initial commit is quite limited in scope. It updates the existing server-side contract tests to send the FDv2 equivalent payloads to SDKs under tests.

These changes do not include **any new tests**. Additional commits are required to expand test coverage to include FDv2 specific handling (e.g. xfer-changes, state handling).

It should also be noted these changes have likely broken the client, mobile, PHP, and Roku integrations. That's okay because we will be updating each of those in subsequent commits as well as the FDv2 work progresses.

### CODE REVIEW

File: framework/types.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
+package framework
```
Comment (keelerm84) [inline framework/types.go:1]: Same basic types I'm carrying around everywhere right now. Need to get these into the events package or somewhere core.

File: mockld/polling_service.go [hunk 2]
```diff
@@ -152,29 +152,84 @@ func (p *PollingService) pollingHandler(getDataFn func(*PollingService, *http.Re
 
 func (p *PollingService) standardPollingHandler() http.Handler {
 	return p.pollingHandler(func(p *PollingService, r *http.Request) []byte {
-		return p.currentData.Serialize()
+		fdv2SdkData, ok := p.currentData.(FDv2SDKData)
```
Comment (keelerm84) [inline mockld/polling_service.go:155]: This translation bit is a mess and is going to require the most rework. I wanted to see how the changes emerged as we developed additional FDv2 tests (like the xfer-changes stuff) before I tried to fix this hack too much.

File: mockld/polling_service.go [hunk 3]
```diff
@@ -152,29 +152,84 @@ func (p *PollingService) pollingHandler(getDataFn func(*PollingService, *http.Re
...
+
+		return data
 	})
 }
 
 func (p *PollingService) phpFlagHandler() http.Handler {
 	return p.pollingHandler(func(p *PollingService, r *http.Request) []byte {
-		data, _ := p.currentData.(ServerSDKData)
-		return data["flags"][mux.Vars(r)["key"]]
+		// TODO: Update this logic
```
Comment (keelerm84) [inline mockld/polling_service.go:213]: PHP stuff will come later.

File: mockld/polling_service_test.go [hunk 4]
```diff
@@ -1,86 +1,84 @@
...
-	doPollingServiceTests(
-		t,
-		ServerSideSDK,
-		EmptyServerSDKData(),
-		NewServerSDKDataBuilder().RawFlag("flag1", json.RawMessage(`{"key": "flag1"}`)).Build(),
-		"GET",
-		"/sdk/latest-all",
-	)
-}
+// TODO: Re-enable these tests
```
Comment (keelerm84) [inline mockld/polling_service_test.go:18]: I will fix the unit tests later. Again, just getting something working out in front of you.

File: mockld/sdk_data.go [hunk 5]
```diff
@@ -41,6 +43,12 @@ type SDKData interface {
 	Serialize() []byte
 }
 
+type FDv2SDKData []framework.BaseObject
```
Comment (keelerm84) [inline mockld/sdk_data.go:46]: This type may have to be more complex over time. We might need something to control the state, the version, and potential the server intent. That way the serialize method can render the full payload and we don't have to do that hacky translation I called out earlier.

File: mockld/sdk_data.go [hunk 6]
```diff
@@ -63,6 +71,23 @@ func (b blockingUnavailableSDKData) Serialize() []byte { return nil }
 // We use this for both regular server-side SDKs and the PHP SDK.
 type ServerSDKData map[DataItemKind]map[string]json.RawMessage
 
+func (s ServerSDKData) AsFDv2SDKData(t *ldtest.T) FDv2SDKData {
```
Comment (keelerm84) [inline mockld/sdk_data.go]: The data model in the parameterized tests used to match the SDK format. That doesn't work anymore. For now, I do a simple translation. Ideally we would do this when loading the files.

**Review by keelerm84 — COMMENTED** [review]
This is a VERY rough first pass at getting the test harness working with FDv2. Lots of todos, but I figure it is better to start getting stuff reviewed and merged so it is accessible to both of us.

File: sdktests/php_eval.go [hunk 7]
```diff
@@ -10,9 +10,9 @@ func doPHPEvalTests(t *ldtest.T) {
 	// how we set up the polling endpoints in the mock data source, and that is handled
 	// transparently by the logic in mockld/polling_service.go based on the fact that the
 	// "SDK kind" configured for test suite is PHP.
-	t.Run("parameterized", runParameterizedServerSideEvalTests)
```
Comment (cwaldren-ld) [inline sdktests/php_eval.go]: Do these deserve TODOs? I feel like it could get lost.

File: sdktests/testapi_sdk_data.go [hunk 8]
```diff
@@ -95,7 +100,9 @@ func NewSDKDataSourceWithoutEndpoint(t *ldtest.T, data mockld.SDKData, options .
 		d.streamingService = mockld.NewStreamingService(data, sdkKind, t.DebugLogger())
 	}
 
-	t.Debug("setting SDK data to: %s", string(data.Serialize()))
+	jsonStr, _ := json.Marshal(data)
```
Comment (cwaldren-ld) [inline sdktests/testapi_sdk_data.go]: could check the error, then use `t` to assert it is nil

**Review by cwaldren-ld — APPROVED** [review]

File: sdktests/php_eval.go (hunk 7 above)
Comment (keelerm84) [reply sdktests/php_eval.go]: I can actually add this back in, so we are good there.

File: sdktests/testapi_sdk_data.go (hunk 8 above)
Comment (keelerm84) [reply sdktests/testapi_sdk_data.go]: Updated!

**Pull request merged**